src/server.h

```diff
@@ -625,6 +625,14 @@ typedef enum { LOG_TIMESTAMP_LEGACY = 0,
 typedef enum { RDB_VERSION_CHECK_STRICT = 0,
                RDB_VERSION_CHECK_RELAXED } rdb_version_check_type;
 
+/* Structure representing a non-owning view of a buffer.
+ * A stringRef struct does not manage the underlying memory, so its destruction
+ * will not free the buffer. */
+typedef struct stringRef {
+    const char *buf; /* Pointer to the externalized buffer */
+    size_t len;      /* Length of the buffer */
+} stringRef;
+
```

Comment:
Please move this type into `entry.c`.  The only other usage is completely unrelated in `db.c` and suggested to be an independent type.  Note this this does not need to be exposed in `entry.h`, it can be completely internal.

Reply:
we kept it here for other occasions we might want to use it (eg scan).
I think I am fine with repeat internal re-implemet this, but why is it so bad to have this type?